src/saturn_engine/worker/topics/rabbitmq.py

```diff
@@ -225,9 +226,26 @@ async def close(self) -> None:
 
     def _serialize(self, message: TopicMessage) -> bytes:
         if self.options.serializer == RabbitMQSerializer.PICKLE:
-            return pickle.dumps(message)
-
-        return json.dumps(asdict(message)).encode()
+            serialized_message = pickle.dumps(message)
+        else:
+            serialized_message = json.dumps(asdict(message)).encode()
+
+        serialized_message_len = len(serialized_message)
+        if self.options.log_above_size:
+            self.logger.warning(
+                "Sending large message",
+                extra={
+                    "data": {
+                        "message": {
+                            "id": message.id,
+                            "size": serialized_message_len,
+                            "tags": message.tags,
```

Comment:
nit, I would omit tags since from the ID we can find the other log (ie. `publishing message`) that will have all the extra details.